src/Schema/OracleSchemaManager.php

```diff
@@ -320,19 +323,226 @@ private function getQuotedIdentifierName($identifier)
+    }
+
+    /**
+     * Gets all foreign keys definitions in one database call.
+     *
+     * @return array<int, array<string, mixed>>
+     */
+    private function getDatabaseForeignKeys(string $databaseName, ?string $tableName = null): array
+    {
+        $sql = <<<'SQL'
+          SELECT COLS.TABLE_NAME,
+                 ALC.CONSTRAINT_NAME,
+                 ALC.DELETE_RULE,
+                 COLS.COLUMN_NAME LOCAL_COLUMN,
+                 COLS.POSITION,
+                 R_COLS.TABLE_NAME REFERENCES_TABLE,
+                 R_COLS.COLUMN_NAME FOREIGN_COLUMN
+            FROM ALL_CONS_COLUMNS COLS
+       LEFT JOIN ALL_CONSTRAINTS ALC ON ALC.OWNER = COLS.OWNER AND ALC.CONSTRAINT_NAME = COLS.CONSTRAINT_NAME
+       LEFT JOIN ALL_CONS_COLUMNS R_COLS ON R_COLS.OWNER = ALC.R_OWNER AND
+                 R_COLS.CONSTRAINT_NAME = ALC.R_CONSTRAINT_NAME AND
+                 R_COLS.POSITION = COLS.POSITION
+SQL;
+
+        $conditions = [];
+        $params     = [];
+
+        $conditions[]    = 'ALC.CONSTRAINT_TYPE = \'R\'';
+        $conditions[]    = 'COLS.OWNER = :OWNER';
+        $params['OWNER'] = $databaseName;
+
+        if ($tableName !== null) {
+            $conditions[]         = 'COLS.TABLE_NAME = :TABLE_NAME';
+            $params['TABLE_NAME'] = $tableName;
+        }
+
+        $sql .= ' WHERE ' . implode(' AND ', $conditions);
+        $sql .= <<<'SQL'
+        ORDER BY COLS.TABLE_NAME, COLS.CONSTRAINT_NAME, COLS.POSITION
+SQL;
+
+        return $this->_conn->fetchAllAssociative($sql, $params);
+    }
+
+    /**
+     * Remove quotes from the given identifier, if needed.
+     *
+     * Uppercases the given identifier if it is not quoted by intention to reflect Oracle's internal auto uppercasing
+     * strategy of unquoted identifiers.
+     */
+    private function unquoteIdentifier(string $identifier): string
+    {
+        $temp = new Identifier($identifier);
+
+        return $temp->isQuoted() ? $temp->getName() : strtoupper($identifier);
+    }
+
     /**
      * {@inheritdoc}
      */
     public function listTableDetails($name): Table
     {
-        $table = parent::listTableDetails($name);
+        $currentDatabase = (string) $this->_conn->getDatabase();
+        $tableName       = $this->unquoteIdentifier($name);
 
-        $sql = $this->_platform->getListTableCommentsSQL($name);
+        $columnsData = $this->getDatabaseColumns($currentDatabase, $tableName);
+        $columns     = $this->_getPortableTableColumnList($name, '', $columnsData);
 
-        $tableOptions = $this->_conn->fetchAssociative($sql);
+        $indexes     = [];
+        $indexesData = $this->getDatabaseIndexes($currentDatabase, $tableName);
+        if (! empty($indexesData)) {
+            $indexes = $this->_getPortableTableIndexesList($indexesData, $name);
+        }
+
+        $foreignKeys     = [];
+        $foreignKeysData = $this->getDatabaseForeignKeys($currentDatabase, $tableName);
+        if (! empty($foreignKeysData)) {
```

Comment:
This is redundant. Please avoid using `empty()` in general since it's too permissive.

Reply:
done